.github/workflows/setup-cluster-credentials.sh

```diff
@@ -1,263 +1,64 @@
 #!/usr/bin/env bash
-################################################
-# This script is invoked by a human who:
-# - has done az login.
-# - can create repository secrets in the github repo from which this file was cloned.
-# - has the gh client >= 2.0.0 installed.
-#
-# This script initializes the repo from which this file is was cloned
-# with the necessary secrets to run the workflows.
-#
-# Script design taken from https://github.com/microsoft/NubesGen.
-#
-################################################
 
-################################################
-# Set environment variables - the main variables you might want to configure.
-#
-# Three letters to disambiguate names.
-DISAMBIG_PREFIX=
-# The location of the resource group. For example `eastus`. Leave blank to use your default location.
-LOCATION=
-RHSM_PASSWORD=
-RHSM_USERNAME=
-RHSM_POOL=
-RHSM_POOL_FOR_RHEL=
-OWNER_REPONAME=
-VM_PASSWORD=
-JBOSS_EAP_USER_PASSWORD=
-SLEEP_VALUE=30s
-# User name for preceding GitHub account.
-USER_NAME=
-# User Email of GitHub acount to access GitHub repository.
-USER_EMAIL=
-# Personal token for preceding GitHub account.
-GIT_TOKEN=
-# Password for database user 'testuser'
-DATABASE_PASSWORD=
+set -e
 
-# End set environment variables
-################################################
+source pre-check.sh
 
+# ANSI color codes
+RED='\033[0;31m'
+NC='\033[0m' # No Color
 
-set -Eeuo pipefail
-trap cleanup SIGINT SIGTERM ERR EXIT
```

Comment:
I see that you have `set -e` elsewhere in this diff. Why have you removed the other set arguments, including `pipefail`? I am curious because my understanding is that `pipefail` is a shell best practice, as documented at https://transang.me/best-practice-to-make-a-shell-script/ .